python/pyspark/cloudpickle/cloudpickle_fast.py

```diff
@@ -631,7 +631,7 @@ def dump(self, obj):
         try:
             return Pickler.dump(self, obj)
         except RuntimeError as e:
-            if "recursion" in e.args[0]:
+            if len(e.args) > 0 and "recursion" in e.args[0]:
```

Comment:
We should also fix this at https://github.com/cloudpipe/cloudpickle. This file is exactly matched with cloudpickle release.

Reply:
Created https://github.com/cloudpipe/cloudpickle/pull/511